inference-engine/include/details/ie_mmap_allocator.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief The header file defines utility PreAllocator class
+ * 
+ * @file ie_pre_allocator.hpp
+ */
+#pragma once
+
+#include <details/ie_exception.hpp>
+#include <memory>
+
+#include "ie_allocator.hpp"
+#include  <mmap/ie_mmap.hpp>
+
+namespace InferenceEngine {
+namespace details {
+/*
+ * @brief This is a helper class to wrap memory mapped files
+ */
+class MmapAllocator : public IAllocator {
+public:
+    using Ptr = std::shared_ptr<MmapAllocator>;
+
+    MmapAllocator(const path_type& path, size_t offset = 0, LockOp lock = LOCK_FOR_READ)
+        : _path(path), _size(0), _offset(offset), _lock(lock) {
+    }
+
+    /**
+     * @brief Locks a handle to heap memory accessible by any memory manipulation routines
+     * @return The generic pointer to a memory buffer
+     */
+    void* lock(void* handle, LockOp) noexcept override {
```

Comment:
should not we check that:
1. `alloc` is called (`_mmap != nullptr`)
2. `handle` is in range `[_mmap->data(), _mmap->data() + _mmap->size()]` ?